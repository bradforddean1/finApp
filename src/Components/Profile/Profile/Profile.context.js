import React from "react";

const ProfileContext = React.createContext({
	name: "n/a",
	logo: "n/a",
	currency: "n/a",
	exchange: "n/a",
	website: "n/a",
	country: "n/a",
	ipo: "n/a",
	current: "n/a",
	change: "n/a",
	changePct: "n/a",
	ticker: "n/a",
	finnhubIndustry: "n/a",
	marketCapitalization: "n/a",
	shareOutstanding: "n/a",
	peNormalizedAnnual: "n/a",
	pbAnnual: "n/a",
	dividendPerShareAnnual: "n/a",
	dividendYieldIndicatedAnnual: "n/a",
	dividendPerShare5Y: "n/a",
	dividendYield5Y: "n/a",
	dividendGrowthRate5Y: "n/a",
	epsBasicExclExtraItemsAnnual: "n/a",
	epsGrowth3Y: "n/a",
	epsGrowth5Y: "n/a",
	epsGrowthTTMYoy: "n/a",
	currentRatioAnnual: "n/a",
	currentRatioQuarterly: "n/a",
});

export default ProfileContext;
