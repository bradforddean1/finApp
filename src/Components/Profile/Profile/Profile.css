.Profile {
	display: flex;
	flex-direction: column;
	padding-top: 4px;
}
