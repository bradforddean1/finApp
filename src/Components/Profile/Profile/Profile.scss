@use "../../../scss/variables" as v;

.Profile {
	display: flex;
	flex-direction: column;
	padding-top: 4px;
	margin-left: v.$gutter;
	margin-right: v.$gutter;
}
