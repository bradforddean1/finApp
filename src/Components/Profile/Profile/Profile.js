import React, { useContext, useState } from "react";
import Header from "../ProfileHeader/ProfileHeader";
import Card from "../ProfileCard/ProfileCard";
import Chart from "../../Chart/Chart";
import ProfileContext from "./Profile.context";
import ButtonAdd from "../../Common/ButtonAdd/ButtonAdd";
import ButtonCta from "../../Common/ButtonCta/ButtonCta";
import Intermodal from "../../Common/Intermodal/Intermodal";
import { authenticatedRequest } from "../../../Utils/api/serverRequest";
import "./Profile.css";

/**
 * Display securities Profile from Profile context
 * @component
 */
function Profile(props) {
	const [showIntermodal, setShowIntermodal] = useState(false);
	const [intermodalContent, setIntermodalContent] = useState();

	const {
		ticker,
		finnhubIndustry,
		marketCapitalization,
		shareOutstanding,
		peNormalizedAnnual,
		pbAnnual,
		dividendPerShareAnnual,
		dividendYieldIndicatedAnnual,
		dividendPerShare5Y,
		dividendYield5Y,
		dividendGrowthRate5Y,
		epsBasicExclExtraItemsAnnual,
		epsGrowth3Y,
		epsGrowth5Y,
		epsGrowthTTMYoy,
		currentRatioAnnual,
		currentRatioQuarterly,
	} = useContext(ProfileContext);

	const handleUndo = () => {
		// setShowIntermodal(false);
		// authenticatedRequest("POST", endpoint, body).then((Response) => {
		// 	if (Response.status === 201) {
		// 	}
		// });
	};

	const addToPortfolio = () => {
		const endpoint = "api/portfolio";
		const body = { ticker: ticker };

		authenticatedRequest("POST", endpoint, body).then((Response) => {
			if (Response.status === 201) {
				setIntermodalContent(
					<>
						<p>{ticker} has been added to your portfolio</p>
						<ButtonCta tag="button" onClick={handleUndo}>
							Undo
						</ButtonCta>
						<ButtonCta tag="a" href="/portfolio">
							Go To Portfolio
						</ButtonCta>
					</>
				);
				setShowIntermodal(true);
			}
		});
	};

	return (
		<div className="Profile panel">
			<Header />
			<Chart />
			<Card
				title="Segment"
				metrics={["Industry", "Market Capitalization", "Shares Outstanding"]}
				values={[finnhubIndustry, marketCapitalization, shareOutstanding]}
			/>
			<Card
				title="Stablity"
				metrics={["Current Ratio Annual", "Current Ratio Quarterly"]}
				values={[currentRatioAnnual, currentRatioQuarterly]}
			/>

			<Card
				title="Value"
				metrics={["PE Ratio", "Price to Book Ratio"]}
				values={[peNormalizedAnnual, pbAnnual]}
			/>
			<Card
				title="Profitability"
				metrics={[
					"Earnigns Per Share",
					"EPS Growth 3Y",
					"EPS Growth 5Y",
					"EPS Growth TTM YOY",
				]}
				values={[
					epsBasicExclExtraItemsAnnual,
					epsGrowth3Y,
					epsGrowth5Y,
					epsGrowthTTMYoy,
				]}
			/>
			<Card
				title="Dividends"
				metrics={[
					"Dividends Per Share Annual",
					"Dividend Yield Indicated Annual",
					"Dividend Per Share 5Y",
					"Dividend Yield 5Y",
					"Dividend Growth Rate 5Y",
				]}
				values={[
					dividendPerShareAnnual,
					dividendYieldIndicatedAnnual,
					dividendPerShare5Y,
					dividendYield5Y,
					dividendGrowthRate5Y,
				]}
			/>
			<ButtonAdd className="add" tag="button" onClick={addToPortfolio} />
			<Intermodal show={showIntermodal} close={() => setShowIntermodal(false)}>
				{intermodalContent}
			</Intermodal>
		</div>
	);
}

export default Profile;
