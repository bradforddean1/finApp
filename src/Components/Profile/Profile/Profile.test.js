import React from "react";
import ReactDOM from "react-dom";
import Profile from "./Profile";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

describe("Quote Component", () => {
	it("Renders", () => {
		const div = document.createElement("div");
		ReactDOM.render(<Profile />, div);
		ReactDOM.unmountComponentAtNode(div);
	});

	it("Renders the Default State", () => {
		const wrapper = shallow(<Profile />);
		expect(toJson(wrapper)).toMatchSnapshot();
	});
});
