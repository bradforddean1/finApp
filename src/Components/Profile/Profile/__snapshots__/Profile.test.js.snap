// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quote Component Renders the Default State 1`] = `
<div
  className="Profile"
>
  <ProfileHeader
    change="n/a"
    changePct="n/a"
    country="n/a"
    currency="n/a"
    current="n/a"
    exchange="n/a"
    ipo="n/a"
    logo="n/a"
    name="n/a"
    ticker="n/a"
    website="n/a"
  />
  <ProfileCard
    metrics={
      Array [
        "Industry",
        "Market Capitalization",
        "Shares Outstanding",
      ]
    }
    title="Segment"
    values={
      Array [
        "n/a",
        "n/a",
        "n/a",
      ]
    }
  />
  <ProfileCard
    metrics={
      Array [
        "Current Ratio Annual",
        "Current Ratio Quarterly",
      ]
    }
    title="Stablity"
    values={
      Array [
        "n/a",
        "n/a",
      ]
    }
  />
  <ProfileCard
    metrics={
      Array [
        "PE Ratio",
        "Price to Book Ratio",
      ]
    }
    title="Value"
    values={
      Array [
        "n/a",
        "n/a",
      ]
    }
  />
  <ProfileCard
    metrics={
      Array [
        "Earnigns Per Share",
        "EPS Growth 3Y",
        "EPS Growth 5Y",
        "EPS Growth TTM YOY",
      ]
    }
    title="Profitability"
    values={
      Array [
        "n/a",
        "n/a",
        "n/a",
        "n/a",
      ]
    }
  />
  <ProfileCard
    metrics={
      Array [
        "Dividends Per Share Annual",
        "Dividend Yield Indicated Annual",
        "Dividend Per Share 5Y",
        "Dividend Yield 5Y",
        "Dividend Growth Rate 5Y",
      ]
    }
    title="Dividends"
    values={
      Array [
        "n/a",
        "n/a",
        "n/a",
        "n/a",
        "n/a",
      ]
    }
  />
  <ButtonAdd
    className="add"
    handleClick={[Function]}
    tag="button"
  />
  <Intermodal
    buttons={Array []}
    close={[Function]}
    show={false}
  />
</div>
`;
